jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/bpmn2/BPMN2DataServiceImpl.java

```diff
@@ -190,18 +191,16 @@ private ProcessAssetDesc fillProcessDefinition(ProcessDescriptor helper, KieCont
 
 	@Override
 	public ProcessDefinition getProcessDefinition(String deploymentId, String processId) {
-	    validateNonEmptyDeploymentIdAndProcessId(deploymentId, processId);
-
-	    if (definitionCache.containsKey(deploymentId)) {
-
-	        ProcessDescriptor helper = definitionCache.get(deploymentId).get(processId);
-	        if (helper == null) {
-	            throw new IllegalStateException("No process available with given id : " + processId);
-	        }
-	        return helper.getProcess();
-	    }
-
-	    return null;
+        if (definitionCache.containsKey(deploymentId)) {
+            ProcessDescriptor helper = definitionCache.get(deploymentId).get(processId);
+            if(helper != null && helper.getProcess() != null) {
+                return helper.getProcess();
+            } else {
+                throw new ProcessDefinitionNotFoundException("No process available with given id : " + processId);
```

Comment:
this requires change in interface description https://github.com/kiegroup/jbpm/blob/master/jbpm-services/jbpm-services-api/src/main/java/org/jbpm/services/api/DefinitionService.java#L62 as it won't return null any more as it is stated there